pkg/frontend/querymiddleware/querysharding_test.go

```diff
@@ -867,22 +867,23 @@ func TestQuerySharding_FunctionCorrectness(t *testing.T) {
 		newSeries(labels.FromStrings("__name__", "bar1", "baz", "blip", "bar", "blop", "foo", "buzz"), start.Add(-lookbackDelta), end, step, factor(8)),
 		newSeries(labels.FromStrings("__name__", "bar1", "baz", "blip", "bar", "blap", "foo", "bazz"), start.Add(-lookbackDelta), end, step, arithmeticSequence(10)),
 	})
-	queryableNativeHistograms := storageSeriesQueryable([]*promql.StorageSeries{
-		newNativeHistogramSeries(labels.FromStrings("__name__", "bar1", "baz", "blip", "bar", "blop", "foo", "barr"), start.Add(-lookbackDelta), end, step, factor(5)),
-		newNativeHistogramSeries(labels.FromStrings("__name__", "bar1", "baz", "blip", "bar", "blop", "foo", "bazz"), start.Add(-lookbackDelta), end, step, factor(7)),
-		newNativeHistogramSeries(labels.FromStrings("__name__", "bar1", "baz", "blip", "bar", "blap", "foo", "buzz"), start.Add(-lookbackDelta), end, step, factor(12)),
-		newNativeHistogramSeries(labels.FromStrings("__name__", "bar1", "baz", "blip", "bar", "blap", "foo", "bozz"), start.Add(-lookbackDelta), end, step, factor(11)),
-		newNativeHistogramSeries(labels.FromStrings("__name__", "bar1", "baz", "blip", "bar", "blop", "foo", "buzz"), start.Add(-lookbackDelta), end, step, factor(8)),
-		newNativeHistogramSeries(labels.FromStrings("__name__", "bar1", "baz", "blip", "bar", "blap", "foo", "bazz"), start.Add(-lookbackDelta), end, step, arithmeticSequence(10)),
-	})
-
 	testQueryShardingFunctionCorrectness(t, queryableFloats, []queryShardingFunctionCorrectnessTest{})
-	testQueryShardingFunctionCorrectness(t, queryableNativeHistograms, []queryShardingFunctionCorrectnessTest{
-		{fn: "histogram_count"},
-		{fn: "histogram_sum"},
-		{fn: "histogram_fraction", tpl: `(<fn>(0,0.5,bar1{}))`},
-		{fn: "histogram_quantile", tpl: `(<fn>(0.5,bar1{}))`},
-	})
+	/*
```

Comment:
Might be good to add a note about why this is commented out

Reply:
I actually don't think this should be commented out.

Reply:
I think it is that case that  @beorn7  made functions more strict in what they return when the input is a histogram. So actually the documentation at https://prometheus.io/docs/prometheus/latest/querying/functions/ might need an update, because now it says: "Functions that do not explicitly mention native histograms in their documentation (see below) effectively treat a native histogram as a float sample of value 0. (This is confusing and will change before native histograms become a stable feature.)" But according to the tests it returns empty set now? Need to verify and probably separate float tests vs histograms.

Reply:
Separated tests for floats and histograms.